File: dotnet/Vuln/Hard/RequestParser.cs
namespace SastEval.Vuln.Hard;

public static class RequestParser
{
    public static UserInputDto Parse(HttpRequest request)
    {
        return new UserInputDto
        {
            Payload = request.Query["payload"].ToString(),
            UserId = request.Query["userId"].ToString(),
            Role = request.Query["role"].ToString()
        };
    }
}

File: java/src/main/java/com/sasteval/vuln/hard/RequestParser.java
package com.sasteval.vuln.hard;

import javax.servlet.http.HttpServletRequest;

/**
 * Utility class that parses HTTP request parameters into a UserInputDto.
 */
public class RequestParser {

    public static UserInputDto parseRequest(HttpServletRequest req) {
        UserInputDto dto = new UserInputDto();
        dto.setPayload(req.getParameter("payload"));
        dto.setUserId(req.getParameter("userId"));
        dto.setRole(req.getParameter("role"));
        return dto;
    }
}
